cmd/start.go

```diff
@@ -91,3 +117,35 @@ func trapSignal(done func()) {
 	// call the cleanup func
 	done()
 }
+
+// UpdateClientsFromChains takes src, dst chains, threshold time and update clients based on expiry time
+func UpdateClientsFromChains(src, dst *relayer.Chain, thresholdTime time.Duration) (time.Duration, error) {
+	var (
+		srcTimeExpiry, dstTimeExpiry time.Duration
+		err                          error
+	)
+
+	eg := new(errgroup.Group)
+	eg.Go(func() error {
+		srcTimeExpiry, err = relayer.GetClientAndUpdate(src, dst, thresholdTime)
+		return err
+	})
+	eg.Go(func() error {
+		dstTimeExpiry, err = relayer.GetClientAndUpdate(dst, src, thresholdTime)
+		return err
+	})
+	if err := eg.Wait(); err != nil {
+		return 0, err
+	}
+
+	if srcTimeExpiry <= 0 || dstTimeExpiry <= 0 {
+		return 0, fmt.Errorf("one of the clients (%s,%s) trusting period is 0 or expired",
+			src.PathEnd.ClientID, dst.PathEnd.ClientID)
+	}
+
+	if srcTimeExpiry <= dstTimeExpiry {
```

Comment:
Can we use [`math.Min`](https://golang.org/pkg/math/#Min)?

Reply:
Actually `math.Min`is using `float64` values but we have `time.Duration` here. I tried to convert them to `float64` but not able to re-convert `float64` value to `time Duration` again after getting min value.

Reply:
[example](https://play.golang.org/p/Lm4xQI_CkJY)

Reply:
I tried like [this](https://play.golang.org/p/e6sDerZdeX9) and it's converting seconds to `ns`. That's the reason, I didn't use `math.Min` . I will update it.

Reply:
Hmm maybe just use nano seconds in math.Min. Then convert back to duration

Reply:
Not a problem. Your method is working fine. I will do it in same way.